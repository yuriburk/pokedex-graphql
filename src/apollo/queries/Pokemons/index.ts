import { gql } from '@apollo/client';

export const GET_POKEMONS = gql`
  query GetPokemons($count: Int!) {
    pokemons(first: $count) {
      id
      image
      name
    }
  }
`;

export const GET_POKEMONS_UPDATED = gql`
  query GetPokemonsUpdated {
    pokemonsUpdated @client
  }
`;
