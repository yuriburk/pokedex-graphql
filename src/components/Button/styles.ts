import styled from 'styled-components';

export const Container = styled.button`
  background: #78c850;
  height: 54px;
  border-radius: 10px;
  border: 0;
  padding: 0 14px;
  width: 100%;
  color: black;
  font-weight: bold;
  margin-top: 16px;
  transition: background-color 0.2 ease 0.2;

  &:hover {
    background: '#78c850';
  }
`;
