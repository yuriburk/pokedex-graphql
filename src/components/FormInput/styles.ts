import styled, { css } from 'styled-components';

import Tooltip from '../Tooltip';

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  flex: 1;
`;

export const Label = styled.label`
  font-weight: bold;
  margin-bottom: 12px;
`;

interface IInputContainerProps {
  isFocused: boolean;
  isFilled: boolean;
  isErrored: boolean;
}

export const InputContainer = styled.div<IInputContainerProps>`
  background: #232129;
  border-radius: 10px;
  padding: 14px;
  border: 2px solid #232129;
  color: ${(props) =>
    props.isFocused || props.isFilled ? '#ff9000' : '#666360'};
  ${(props) =>
    props.isErrored &&
    css`
      border-color: #c53030;
    `}
  ${(props) =>
    props.isFocused &&
    css`
      border-color: #ff9000;
    `}

  display: flex;
  align-items: center;

  & + div {
    margin-top: 12px;
  }

  input {
    flex: 1;
    background: transparent;
    border: 0;
    color: #f4ede8;
    width: 100%;

    &::placeholder {
      color: #666360;
    }

    &:-webkit-autofill,
    &:-webkit-autofill:hover,
    &:-webkit-autofill:focus,
    &:-webkit-autofill:active {
      -webkit-box-shadow: 0 0 0 50px #232129 inset;
      -webkit-text-fill-color: #f4ede8;
    }
  }

  svg {
    margin-right: 16px;
  }
`;

export const Error = styled(Tooltip)`
  height: 20px;
  margin-left: 16px;

  svg {
    margin: 0;
  }

  span {
    background: #c53030;
    color: #fff;

    &::before {
      border-color: #c53030 transparent;
    }
  }
`;
