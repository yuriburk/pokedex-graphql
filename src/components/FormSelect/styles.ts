import { CSSProperties } from 'react';
import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  flex: 1;

  margin-bottom: 8px;
`;

export const Label = styled.label`
  font-weight: bold;
  margin-bottom: 2px;
`;

export const customStyles = {
  container: (base: CSSProperties): CSSProperties => ({
    ...base,
    flex: 1,
  }),
  control: (base: CSSProperties): Record<string, unknown> => ({
    ...base,
    backgroundColor: '#232129',
    borderColor: '#232129',
    padding: '6px',
    boxShadow: 'none',
    borderRadius: '10px',
    '&:hover': {
      borderColor: '#ff9000',
    },
  }),
  multiValue: (base: CSSProperties): CSSProperties => ({
    ...base,
    backgroundColor: '#1d1c24',
  }),
  multiValueLabel: (base: CSSProperties): CSSProperties => ({
    ...base,
    color: '#fff',
  }),
  menuList: (base: CSSProperties): CSSProperties => ({
    ...base,
    color: '#fff',
    backgroundColor: '#232129',
  }),
  option: (base: CSSProperties): Record<string, unknown> => ({
    ...base,
    backgroundColor: '#232129',
    '&:hover': {
      backgroundColor: '#1d1c24',
    },
  }),
  singleValue: (base: CSSProperties): Record<string, unknown> => ({
    ...base,
    color: '#fff',
  }),
};
