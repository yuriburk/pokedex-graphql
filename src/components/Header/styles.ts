import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-bottom: 18px;

  svg {
    cursor: pointer;
  }
`;
