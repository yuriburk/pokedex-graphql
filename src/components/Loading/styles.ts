import styled from 'styled-components';

export const LoadingContainer = styled.div`
  position: fixed;
  top: 50%;
  left: 50%;
`;
