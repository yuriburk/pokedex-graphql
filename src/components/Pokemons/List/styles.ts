import styled, { css } from 'styled-components';
import { IPokemonTypes } from 'interfaces';

export const Container = styled.div``;

export const List = styled.ul`
  display: flex;
  justify-content: flex-start;
  flex-wrap: wrap;

  list-style: none;
`;

export const ListItem = styled.li`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  flex: 1;

  cursor: pointer;
  min-height: 289px;
  padding: 14px 30px;
  border-radius: 8px;
  margin: 4px;
  background-color: ${(props) => props.theme.colors.secundaryDark};

  transition: transform ease 0.5s;

  &:hover {
    transform: translateY(-6px);
  }
`;

export const PokemonInfo = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;

export const Title = styled.h2`
  color: ${(props) => props.theme.colors.primaryWhite};
  margin-top: 8px;
  word-break: break-word;
  overflow: hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
`;

export const TextInfo = styled.p`
  color: white;
`;

export const SpecialContainer = styled.div`
  display: flex;
  justify-content: center;
  flex-wrap: wrap;

  margin-bottom: 22px;
`;

interface ISpecialInfoContainerProps {
  pokemonType: IPokemonTypes;
}

export const SpecialInfoContainer = styled.div<ISpecialInfoContainerProps>`
  padding: 4px 8px;
  margin: 4px 4px 0 0;
  border-radius: 12px;
  ${(props) => props.theme.typesColors[props.pokemonType]};
`;

export const SpecialInfo = styled.p`
  color: ${(props) => props.theme.colors.primaryWhite};
`;

export const Image = styled.img`
  width: 124px;
  height: 124px;
  border-radius: 8px;
`;

export const ButtonContainer = styled.div`
  display: flex;
  justify-content: flex-end;
`;
