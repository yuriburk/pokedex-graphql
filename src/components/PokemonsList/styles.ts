import styled from 'styled-components';

export const Container = styled.ul`
  display: flex;
  flex-wrap: wrap;

  list-style: none;
`;

export const Item = styled.li`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  flex: 1;

  padding: 15px;
  border-radius: 5px;
  margin: 5px;

  background-color: white;
`;

export const Image = styled.img`
  width: 125px;
  height: 125px;
`;

export const Title = styled.h2`
  color: black;

  margin-top: 15px;
`;
