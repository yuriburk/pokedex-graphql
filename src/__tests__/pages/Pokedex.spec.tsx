import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { MockedProvider } from '@apollo/client/testing';
import { ThemeProvider } from 'styled-components';

import { theme } from 'styles/theme';
import Pokedex from 'pages/Pokedex';
import { IPokemon } from 'interfaces';
import { GET_POKEMONS_CACHED } from 'operations/queries/Pokemons/cache';

jest.mock('react-router-dom', () => ({
  useHistory: () => ({
    push: jest.fn(),
  }),
  useParams: jest.fn(() => ({ id: 'test-id' })),
  Link: ({ children }: { children: React.ReactNode }) => children,
}));

const pokemon: IPokemon = {
  id: 'id-1',
  name: 'Pikachu',
  image: 'image',
  number: '001',
  weaknesses: ['Rock'],
  classification: 'Mouse PokÃ©mon',
  maxCP: 100,
  maxHP: 400,
  resistant: ['Water'],
  attacks: {
    special: [
      {
        name: 'Thunder',
        damage: 90,
        type: 'Electric',
      },
    ],
  },
  weight: {
    minimum: '0.1kg',
    maximum: '4kg',
  },
  height: {
    minimum: '0.1m',
    maximum: '0.5m',
  },
  evolutions: [
    {
      id: 'id-2',
    },
  ],
};

const mocks = [
  {
    request: {
      query: GET_POKEMONS_CACHED,
      variables: {
        count: 151,
      },
    },
    result: {
      data: {
        pokemons: [pokemon],
      },
    },
  },
];

const PokedexWrapper = () => (
  <MockedProvider mocks={mocks} addTypename={false}>
    <ThemeProvider theme={theme}>
      <Pokedex />
    </ThemeProvider>
  </MockedProvider>
);

describe('Pokedex Component', () => {
  afterEach(cleanup);

  it('should be able to render without errors', () => {
    const { container } = render(<PokedexWrapper />);

    expect(container).toMatchSnapshot();
  });
});
