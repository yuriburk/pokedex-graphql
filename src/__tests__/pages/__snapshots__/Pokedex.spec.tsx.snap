// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokedex Page should be able to render without errors 1`] = `
<div>
  <div
    class="sc-fznKkj lbPMLS"
  >
    <div
      class="sc-fznZeY heAsA-d"
    >
      <h1>
        Pok√©dex
      </h1>
      <div
        class="sc-fznyAO gwRYLQ"
        style="width: 240px; height: 48px;"
      >
        <input
          placeholder="Search"
        />
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="11"
            cy="11"
            r="8"
          />
          <line
            x1="21"
            x2="16.65"
            y1="21"
            y2="16.65"
          />
        </svg>
      </div>
    </div>
    <div
      class="sc-AxjAm StDqN"
    >
      <ul
        class="sc-AxirZ chGCDH"
      >
        <li
          class="sc-AxiKw hYtiYe"
          data-testid="list-item-Pikachu"
        >
          <div
            class="sc-AxhCb fjiAYs"
          >
            <img
              alt="Pikachu"
              class="sc-fzozJi ubzuZ"
              src="image"
            />
            <h2
              class="sc-AxhUy SGDKj"
            >
              Pikachu
            </h2>
            <p
              class="sc-AxgMl dccyOx"
            >
              #
              001
            </p>
          </div>
          <div
            class="sc-AxheI CIhFB"
          >
            <div
              class="sc-Axmtr kTUKtH"
            >
              <p
                class="sc-AxmLO irKRQn"
              >
                Electric
              </p>
            </div>
          </div>
        </li>
        <li
          class="sc-AxiKw hYtiYe"
          data-testid="list-item-Raichu"
        >
          <div
            class="sc-AxhCb fjiAYs"
          >
            <img
              alt="Raichu"
              class="sc-fzozJi ubzuZ"
              src="image"
            />
            <h2
              class="sc-AxhUy SGDKj"
            >
              Raichu
            </h2>
            <p
              class="sc-AxgMl dccyOx"
            >
              #
              002
            </p>
          </div>
          <div
            class="sc-AxheI CIhFB"
          >
            <div
              class="sc-Axmtr hguHGR"
            >
              <p
                class="sc-AxmLO irKRQn"
              >
                Normal
              </p>
            </div>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
